pynest/nest/server/hl_api_server.py

```diff
@@ -247,16 +251,28 @@ def get_arguments(request):
     return list(args), kwargs
 
 
-def get_globals():
-    """Get globals for exec function."""
-    copied_globals = globals().copy()
+def get_modules_from_env():
+    """Get modules from environment variable NEST_SERVER_MODULES.
 
-    # Add modules to copied globals
-    modlist = [(module, importlib.import_module(module)) for module in MODULES]
-    modules = dict(modlist)
-    copied_globals.update(modules)
+    This function converts the content of the environment variable NEST_SERVER_MODULES:
+    to a formatted dictionary for updating the Python `globals`.
 
-    return copied_globals
+    Here is an example:
+        `NEST_SERVER_MODULES="nest,numpy as np,random from numpy"`
+    is converted to the following dictionary:
+        `{'nest': <module 'nest'> 'np': <module 'numpy'>, 'random': <module 'numpy.random'>}`
+    """
+    modules = {}
+    for module in MODULES:
+        if " as " in module:
+            modname, modvar = module.split(" as ")
+            modules[modvar] = importlib.import_module(modname)
+        elif " from " in module:
+            modvar, modname = module.split(" from ")
+            modules[modvar] = importlib.import_module(f"{modname}.{modvar}")
+        else:
+            modules[module] = importlib.import_module(module)
+    return modules
```

Comment:
What about an AST approach? We'd avoid introducing our own syntax, support the all Python importing language features, and we don't need to parse anything ourselves? 

```
import nest; import numpy as np; from numpy import random
```

```suggestion
    Here is an example:
        `NEST_SERVER_MODULES="import nest, numpy as np; from numpy import random"`
    is converted to the following dictionary:
        `{'nest': <module 'nest'> 'np': <module 'numpy'>, 'random': <module 'numpy.random'>}`
    """
    modules = {}
    try:
        parsed = ast.iter_child_nodes(ast.parse(MODULES))
    except SyntaxError, ValueError:
        raise SyntaxError("The NEST server module environment variables contains syntax errors.")
    for node in parsed:
        if isinstance(node, ast.Import):
            for alias in node.names:
              modules[alias.asname or alias.name] = importlib.import_module(alias.name)
        elif isinstance(node, ast.ImportFrom):
            for alias in node.names:
              modules[alias.asname or alias.name] = importlib.import_module(f"{node.module}.{alias.name}")
    return modules
```

Reply:
Many thanks! Your code seems more solid. 

Can you please check line 261 to this? 
```
`NEST_SERVER_MODULES="import nest; import numpy as np; from numpy import random"`
```

Reply:
done :)